src/FluentValidation/Internal/Extensions.cs

```diff
@@ -200,5 +199,27 @@ public static class Extensions {
 
 			return false;
 		}
+
+		private static Func<object, TResult> GetSafeValue<T, TProperty, TResult>(Func<T, TProperty> func) where TProperty : TResult {
+			return x => {
+				try {
+					return func((T)x);
+				}
+				catch (Exception) {

```

Comment:
We shouldn't be catching general `Exception` here. Can you catch a more specific exception type?

Reply:
I have sent a new commit to catch only `NullReferenceException`.